enhancements/baremetal/continuous-hw-introspection.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Continuous hardware introspecton of Baremetal IPI hosts
+authors:
+  - "@sadasu"
+reviewers:
+  - "@dhellmann"
+  - "@enxebre"
+  - "@hardys"
+  - "@stbenjam"
+  - "@juliakreger"
+approvers:
+  - "@dhellmann"
+  - "@enxebre"
+creation-date: 2020-02-18
+last-updated: 2020-02-17
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+---
+
+# Continuous hardware introspection of Baremetal IPI hosts
+
+This proposal talks about how performing continuous hardware introspection on all master
+and worker nodes in a Baremetal IPI deployment would get us important data about baremetal
+hosts which can be used to monitor changes in hardware capabilities of the server.
+
+This enhancement will not result in a new CRD or API to be written but will detail
+data that would become availabe in the Status field of the BareMetalHost CRD.
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Continuous hardware inspection on all the master and worker nodes that are a part
+of a Baremetal IPI deployment, allow the user to detect changes in server hardware
+and become aware of any failures seen on the servers. This information can be used
+to report current status of hardware to the user and/or glean metrics about the
+health of the deployment.
+
+At the end of 4.4, a Baremetal IPI deployment does not have continuous hardware introspection
+running on its master or worker nodes. The underlying provisioning service already has the
+capability to perform hardware inspection on each of the servers periodically. The
+baremetal-operator (BMO) then reads this information and updates the "status" field of the
+BareMetalHost CRD. Please refer to [1] for the latest BaremetalHost CRD.
+
+## Motivation
+
+This enhancement will allow the user to get continuous updates on all the baremetal
```

Comment:
can we add some real world scenarios here as user stories i.e "As a user I'd like to visualise when..."

Reply:
Done.